components/content_settings/core/browser/brave_content_settings_pref_provider.cc

```diff
@@ -22,30 +22,35 @@
 #include "components/prefs/pref_service.h"
 #include "content/public/browser/browser_task_traits.h"
 #include "content/public/browser/browser_thread.h"
+#include "net/base/registry_controlled_domains/registry_controlled_domain.h"
 
 namespace content_settings {
 
 namespace {
 
 Rule CloneRule(const Rule& rule, bool reverse_patterns = false) {
-  auto secondary_pattern = rule.secondary_pattern;
-  if (secondary_pattern ==
+  // brave plugin rules incorrectly use first party url as primary
+  auto primary_pattern = reverse_patterns ? rule.secondary_pattern
+                                          : rule.primary_pattern;
+  auto secondary_pattern = reverse_patterns ? rule.primary_pattern
+                                            : rule.secondary_pattern;
+
+  if (primary_pattern ==
       ContentSettingsPattern::FromString("https://firstParty/*")) {
-    if (!rule.primary_pattern.MatchesAllHosts()) {
-      secondary_pattern = ContentSettingsPattern::FromString(
-          "*://[*.]" + rule.primary_pattern.GetHost() + "/*");
+    DCHECK(reverse_patterns);  // we should only hit this for brave plugin rules
```

Comment:
I guess we still run tests without `DCHECK`s - mb just for curiosity add `DumpWithoutCrashing()` here

Reply:
I ran locally with dcheck enabled